lib/src/element_type.dart

```diff
@@ -274,32 +292,38 @@ abstract class DefinedElementType extends ElementType {
   Class get boundClass =>
       ModelElement.fromElement(_boundClassElement, packageGraph);
 
-  InterfaceType get _interfaceType => type;
+  DartType get _bound => type;
 
-  InterfaceType _instantiatedType;
+  DartType _instantiatedType;
 
   /// Return this type, instantiated to bounds if it isn't already.
+  @override
   DartType get instantiatedType {
     if (_instantiatedType == null) {
-      if (!_interfaceType.typeArguments.every((t) => t is InterfaceType)) {
+      if (_bound is InterfaceType &&
+          !(_bound as InterfaceType)
+              .typeArguments
+              .every((t) => t is InterfaceType)) {
         var typeSystem = library.element.typeSystem as TypeSystemImpl;
-        _instantiatedType = typeSystem.instantiateToBounds(_interfaceType);
+        _instantiatedType = typeSystem.instantiateToBounds(_bound);
```

Comment:
Please use `ClassElement.instantiateToBounds` if it is available in your version of analyzer.

Reply:
Will migrate in a separate PR.